src/components/MainNav/MyProfile/MyProfile.js

```diff
@@ -47,6 +55,38 @@ export default class MyProfile extends Component {
     this.getProfileImage = this.getProfileImage.bind(this);
   }
 
+  componentDidMount() {
+    const { modules } = this.props;
+    const navLinks = modules.settings.map(({ links }) => links);
+
+    if (navLinks && Array.isArray(navLinks)) {
+      const links = navLinks[0].userDropdown;
+      links.forEach((link, index) => {
+        if (this.props.stripes.user) {
+          if (!link.check) {
+            this.createLink(link, index);
+          } else if (typeof userDropodownChecks[link.check] === 'function') {
+            userDropodownChecks[link.check]().then((canShow) => {
```

Comment:
Do we really need an async API with promises here? Why not just a simple function returning a boolean?

Reply:
This call is reasonable in order to check that the user has a local password. Without an async call, there is no way to check this.